trace/tracestate.go

```diff
@@ -39,43 +33,102 @@ const (
 	errDuplicate     errorConst = "duplicate list-member in tracestate"
 )
 
-var (
-	noTenantKeyRe   = regexp.MustCompile(`^` + noTenantKeyFormat + `$`)
-	withTenantKeyRe = regexp.MustCompile(`^` + withTenantKeyFormat + `$`)
-	valueRe         = regexp.MustCompile(`^` + valueFormat + `$`)
-	memberRe        = regexp.MustCompile(`^\s*((?:` + noTenantKeyFormat + `)|(?:` + withTenantKeyFormat + `))=(` + valueFormat + `)\s*$`)
-)
-
 type member struct {
 	Key   string
 	Value string
 }
 
-func newMember(key, value string) (member, error) {
-	if len(key) > 256 {
-		return member{}, fmt.Errorf("%w: %s", errInvalidKey, key)
+func checkValueChar(v byte) bool {
+	// [\x20-\x2b\x2d-\x3c\x3e-\x7e]*
+	return v >= '\x20' && v <= '\x7e' && v != '\x2c' && v != '\x3d'
+}
+
+func checkValueLast(v byte) bool {
+	// [\x21-\x2b\x2d-\x3c\x3e-\x7e]
+	return v >= '\x21' && v <= '\x7e' && v != '\x2c' && v != '\x3d'
+}
+
+func checkValue(val string) bool {
+	n := len(val)
+	if n == 0 || n > 256 {
+		return false
 	}
-	if !noTenantKeyRe.MatchString(key) {
-		if !withTenantKeyRe.MatchString(key) {
-			return member{}, fmt.Errorf("%w: %s", errInvalidKey, key)
+	// valueFormat         = `[\x20-\x2b\x2d-\x3c\x3e-\x7e]{0,255}[\x21-\x2b\x2d-\x3c\x3e-\x7e]`
+	for i := 0; i < n-1; i++ {
+		if !checkValueChar(val[i]) {
+			return false
 		}
-		atIndex := strings.LastIndex(key, "@")
-		if atIndex > 241 || len(key)-1-atIndex > 14 {
-			return member{}, fmt.Errorf("%w: %s", errInvalidKey, key)
+	}
+	return checkValueLast(val[n-1])
+}
+
+func checkKeyRemain(key string) bool {
+	// [_0-9a-z\-\*\/]*
+	for _, v := range key {
+		if (v >= '0' && v <= '9') || (v >= 'a' && v <= 'z') {
+			continue
+		}
+		switch v {
+		case '_', '-', '*', '/':
+			continue
 		}
+		return false
 	}
-	if len(value) > 256 || !valueRe.MatchString(value) {
+	return true
+}
+
+func checkKeyPart(key string, n int) bool {
```

Comment:
Please add function documentation. Be sure to include information about what `n` is in this call.